src/java.base/share/classes/jdk/internal/util/random/RandomWrapper.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.internal.util.random;
+
+import java.util.Random;
+import java.util.random.RandomGenerator;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+
+/**
+ * Class used to wrap a {@link java.util.random.RandomGenerator} to
+ * {@link java.util.Random}
+ */
+
+@SuppressWarnings("serial")
+public class RandomWrapper extends Random implements RandomGenerator {
+    private final RandomGenerator randomToWrap;
+
+    private RandomWrapper(RandomGenerator randomToWrap) {
+        this.randomToWrap = randomToWrap;
+    }
+
+    public static Random wrapRandom(RandomGenerator random) {
+        // Check to see if its not wrapping another Random instance
+        if (random instanceof Random)
+            return (Random) random;
+
+        return new RandomWrapper(random);
+    }
+
+    /**
+     * setSeed does not exist in {@link java.util.random.RandomGenerator} so can't
+     * use it
+     */
```

Comment:
Its debatable whether setSeed can ignore the argument or should throw an exception.
If it ignores the argument, it is explicitly violating the contract of the supertype Random.setSeed
that says the seed is set.  
I'd recommend throwing an UnsupportedOperationException because setSeed will not work as expected.
Other classes (ThreadLocalRandom) throws UnsupportedOperationException from setSeed.

Reply:
Yes, probably following ThreadLocalRandom is the right thing. Pretty clunky but I think it's necessary.

Reply:
ok so i made the change to check if its initialized and throw if the user tries to use setSeet() c297d42f4505b03780bd9ab2e169546fd6cffcb9